external/include/error_report/error_report.h

```diff
@@ -0,0 +1,101 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#ifndef ERROR_REPORT_H
+#define ERROR_REPORT_H
+
+#include <tinyara/config.h>
+#include <time.h>
+#include <stdint.h>
+
+/**
+ * @defgroup error_report error_report
+ * @ingroup error_report
+ * @brief Provides APIs for Error Reporting Modules
+ * @{
+ */
+#define ERR_FUNCNAME_LEN_MAX	32
+#define ERR_TASKNAME_LEN_MAX CONFIG_TASK_NAME_SIZE + 1
+#define ERR_TIMESTAMP_LEN	26
+#define ERR_BUFLEN	1024
+enum _rt_error_module_id {
```

Comment:
Usually starting with ```_``` means it is for internal. Why do you define it in public header?